File: backend/schemas/technology_intelligence.py
"""Technology Intelligence response schema.

Mirrors backend/services/technology_intelligence_service.py::describe().

Every field beyond the identity is *evidence*, not decoration. The
lifecycle label alone would ask a user to trust a verdict; the counters
and the evidence sentence let them check it, which matters more here than
elsewhere because this replaced a feature that was confidently wrong.
"""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class TechnologyObservation(BaseModel):
    source: str
    observed_at: str
    research_session_id: Optional[str] = None


class TechnologyIntelligenceOut(BaseModel):
    id: str
    company_id: str
    name: str
    category: Optional[str] = None

    lifecycle: str
    lifecycle_label: str
    # Carried in the payload rather than mapped in the frontend so the
    # careful wording - particularly that "not observed recently" is not
    # evidence of removal - cannot be lost in translation to a surface
    # that never read the service.
    lifecycle_description: str

    first_seen_at: Optional[datetime] = None
    last_seen_at: Optional[datetime] = None
    observation_count: int
    missed_count: int
    consecutive_misses: int
    # The observation rate, not a tuned score: observations / times looked.
    confidence: float
    evidence_summary: str
    observation_sources: list = []

File: frontend/react/src/types/technologyIntelligence.ts
// Technology Intelligence (mirrors backend/schemas/technology_intelligence.py).
//
// The lifecycle *label and description* come from the backend rather than
// being mapped here. That is deliberate: the wording carries a claim
// Scout must not overstate - particularly that "not observed recently" is
// not evidence a company stopped using something - and a frontend that
// invented its own phrasing could quietly lose that care.

export type TechnologyLifecycle = "newly_detected" | "emerging" | "established" | "stale";

export interface TechnologyObservation {
  source: string;
  observed_at: string;
  research_session_id: string | null;
}

export interface TechnologyIntelligence {
  id: string;
  company_id: string;
  name: string;
  category: string | null;

  lifecycle: TechnologyLifecycle;
  lifecycle_label: string;
  lifecycle_description: string;

  first_seen_at: string | null;
  last_seen_at: string | null;
  observation_count: number;
  missed_count: number;
  consecutive_misses: number;
  // The observation rate, not a tuned score: observations / times looked.
  confidence: number;
  evidence_summary: string;
  observation_sources: TechnologyObservation[];
}
